libs/geoservices/covid/src/lib/pages/maps/components/lockdown-map/lockdown-map.component.ts

```diff
@@ -0,0 +1,258 @@
+          paint: {
+            'fill-outline-color': '#FFFFFF'
+          }
+        },
+        'counties-lockdowns'
+      );
+
+      map.addLayer(
+        {
+          id: 'statelines',
+          type: 'line',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          paint: {
+            'line-color': '#000000',
+            'line-width': 1
+          }
+        },
+        'state-label'
+      );
+
+      map.on('mousemove', 'counties-lockdowns', (e) => {
+        const feature = e.features[0];
+        this.countyBoxModel.next({
+          county: feature.properties.NAME
+        });
+      });
+
+      map.on('click', 'counties-lockdowns', (e) => {
+        const feature = e.features[0];
+        const hoveredCounty = this.statData.filter((county) => county.fips === feature.properties.fips);
+        const updatedDate: Date = hoveredCounty[0]['lockdownInfo'][0]
+          ? hoveredCounty[0]['lockdownInfo'][0]['updated']
+          : null;
+        const createdDate: Date = hoveredCounty[0]['lockdownInfo'][0]
+          ? hoveredCounty[0]['lockdownInfo'][0]['created']
+          : null;
+        this.infoBoxModel.next({
+          name: feature.properties.NAME,
+          state: hoveredCounty[0]['state'],
+          fips: hoveredCounty[0]['fips'],
+          fipsNum: hoveredCounty[0]['fipsNum'],
+          claims: hoveredCounty[0]['claims'],
+          sites: hoveredCounty[0]['sites'],
+          lockdowns: hoveredCounty[0]['lockdowns'],
+          lockdownInfo: hoveredCounty[0]['lockdownInfo'],
+          updated: updatedDate,
+          created: createdDate
+        });
+      });
+
+      map.dragRotate.disable();
+    });
+  }
+
+  public getFipsLockdownData(fipsCode: string) {
+    const lockdownUrl = 'https://nodes.geoservices.tamu.edu/api/covid/lockdowns/active/county/' + fipsCode;
+    const requests = forkJoin([this.http.get(lockdownUrl)]);
+    requests.pipe(take(1)).subscribe(([lockdownData]) => {
+      console.log(lockdownData);
+      return lockdownData;
+    });
+  }
+
+  public getStats(): void {
+    const statsUrl = 'https://nodes.geoservices.tamu.edu/api/covid/counties/stats';
+    const requests = forkJoin([this.http.get(statsUrl)]);
+    requests.pipe(take(1)).subscribe(([recievedData]) => {
+      this.unformattedData = recievedData;
+      this.formatData();
+      this.drawClaimsMap();
+      this.drawLockdownMap();
+    });
+  }
+
+  public formatData(): void {
+    this.statData = new Array();
+    let lockdownRecords = new Array();
+    Object.keys(this.unformattedData).forEach((keyIndex) => {
+      lockdownRecords = this.unformattedData[keyIndex]['lockdownInfo'].filter((item) => {
+        return item !== null;
+      });
+      lockdownRecords = lockdownRecords.map((item) => {
+        const lockdownStat: LockdownRecord = {
+          updated: item.updated,
+          created: item.created,
+          guid: item.guid
+        };
+        return lockdownStat;
+      });
```

Comment:
You can chain these the `map` calls.

Example:

```js
const someArray = [0, 2, 3, undefined, 4, 5, 6];

console.log(someArray.filter(n => n !== undefined).map(n => n + 1));

// 1, 3, 4, 5, 6, 7
```

Reply:
got it chained